pallets/swaps/src/lib.rs

```diff
@@ -273,36 +290,33 @@ pub mod pallet {
 			who: &T::AccountId,
 			swap_id: Self::SwapId,
 			from_currency: Self::CurrencyId,
-		) -> Result<Self::Amount, DispatchError> {
+		) -> Result<T::BalanceOut, DispatchError> {
 			Ok(SwapIdToOrderId::<T>::get((who, swap_id))
-				.and_then(T::OrderBook::get_order_details)
+				.and_then(T::OrderBook::<T::BalanceIn, T::BalanceOut>::get_order_details)
 				.filter(|swap| swap.currency_out == from_currency)
 				.map(|swap| swap.amount_out)
-				.unwrap_or_default())
+				.unwrap_or(Zero::zero()))
 		}
 
 		fn valid_pair(currency_in: Self::CurrencyId, currency_out: Self::CurrencyId) -> bool {
-			T::OrderBook::valid_pair(currency_in, currency_out)
+			T::OrderBook::<T::BalanceIn, T::BalanceOut>::valid_pair(currency_in, currency_out)
```

Comment:
Good question! Because `valid_pair()` does not receive or return any balance (it's independent of the balance), you can basically put there what you want and it will compile and work.

Actually, because it's independent, the resulting patter indirectly "suggests" that a better design might be to move `valid_pair()` out of the trait, which semantically makes sense, I think.